File: src/main/java/com/ninja/ieumnori/ieumnoriserver/magazine/domain/Magazine.java
package com.ninja.ieumnori.ieumnoriserver.magazine.domain;

import com.ninja.ieumnori.ieumnoriserver.common.domain.BaseEntity;
import com.ninja.ieumnori.ieumnoriserver.simulation.domain.Simulation;

import jakarta.persistence.*;
import lombok.*;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
@Entity
@Table(name = "magazine")
public class Magazine extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Builder.Default
    @Column(name = "is_active", nullable = false)
    private boolean isActive = true;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "simulation_id", nullable = false)
    private Simulation simulation;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", nullable = false)
    private MagazineCategory category;

    @Column(name = "main_title", nullable = false)
    private String mainTitle;

    @Column(name = "sub_title")
    private String subTitle;

    @Column(name = "thumbnail_image_url")
    private String thumbnailImageUrl;

    @Lob
    @Column(name = "content")
    private String content;

    @Builder.Default
    @Column(name = "view_count", nullable = false)
    private int viewCount = 0;

    public void increaseViewCount() {
        this.viewCount++;
    }

    public void update(String mainTitle, String subTitle, String thumbnailImageUrl, String content) {
        this.mainTitle = mainTitle;
        this.subTitle = subTitle;
        this.thumbnailImageUrl = thumbnailImageUrl;
        this.content = content;
    }
}

File: src/main/java/com/ninja/ieumnori/ieumnoriserver/magazine/domain/Magazine.kt
package com.ninja.ieumnori.ieumnoriserver.magazine.domain

import com.ninja.ieumnori.ieumnoriserver.common.domain.BaseEntity
import jakarta.persistence.*

@Entity
@Table(name = "magazine")
class Magazine (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
    @Column(name="main_title", nullable = false)
    var mainTitle: String,
    @Column(name="sub_title", nullable = false)
    var subTitle: String,
    @Column(name="main_image", nullable = false)
    var mainImage: String,
    @Column(name="category_id", nullable = false)
    var categoryId: Long,
    @Transient // DB 컬럼에 반영되지 않음
    var cardNewsIdList: List<Long> = emptyList(),
    @Column(nullable = false)
    var content: String,
    @Column(name = "is_exposed", nullable = false)
    var isExposed: Boolean,

): BaseEntity() {
    constructor(): this(0, "", "", "", 0, emptyList<Long>(), "", false)
}